master/internal/experiment.go

```diff
@@ -484,25 +504,28 @@ func (e *experiment) isBestValidation(msg searcher.CompletedMessage) bool {
 	return isBest
 }
 
-func (e *experiment) updateState(ctx *actor.Context, state model.State) {
+func (e *experiment) updateState(ctx *actor.Context, state model.State) (ok bool) {
 	if wasPatched, err := e.Transition(state); err != nil {
 		ctx.Log().Error(err)
-		return
+		return false
 	} else if !wasPatched {
-		return
+		return true
 	}
 	telemetry.ReportExperimentStateChanged(ctx.Self().System(), e.db, *e.Experiment)
 
 	ctx.Log().Infof("experiment state changed to %s", state)
 	for _, child := range ctx.Children() {
 		ctx.Tell(child, state)
 	}
+
 	if err := e.db.SaveExperimentState(e.Experiment); err != nil {
 		ctx.Log().Error(err)
 	}
 	if e.canTerminate(ctx) {
 		ctx.Self().Stop()
 	}
+	// "Long story short, we never put effort into db error handling".
```

Comment:
non-blocking: this comment, while applicable, does not offer a lot of context in this location. It is probably better removed than floating at the end of this function.

Reply:
I updated the comment. I think it's worthwhile to leave it here to avoid confusing readers into thinking this is an oversight.